### PR TITLE
Membership requests

### PR DESCRIPTION
**Pull request by HSalmon**
- Need to use MembershipRequestsService in 'approve' controller action
- Need to extend MembershipRequestsService.approve to authorize with CanCan

### CODE REVIEW

File: app/controllers/api/membership_requests_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class API::MembershipRequestsController < API::RestfulController
+
+  def pending
+    load_and_authorize :group
```
Comment (robguthrie) [inline app/controllers/api/membership_requests_controller.rb]: @HSalmon - another note! - I think we should be checking if the user can manage_membership_requests for this group before revealing pending and previous requests.

**Comment by HSalmon** [comment]
@robguthrie, @gdpelican, failure is unrelated. Still needs some styling but wondering whether that could be applied later in the week?

screen shot 2015-07-06 at 4 29 07 pm

File: lineman/app/components/membership_requests_page/membership_requests_page_controller.coffee [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  @approve = (membershipRequest) =>
+    Records.membershipRequests.approve(membershipRequest).then ->
+      FlashService.success "membership_requests_page.messages.request_approved_success"
+
+
+  @ignore = (membershipRequest) =>
+    Records.membershipRequests.ignore(membershipRequest).then ->
+      FlashService.success "membership_requests_page.messages.request_ignored_success"
+
+  @noPendingRequests = =>
```
Comment (HSalmon) [inline lineman/app/components/membership_requests_page/membership_requests_page_controller.coffee]: This function doesn't seem to work - had to take it out of the view...

File: lineman/app/components/membership_requests_page/membership_requests_page_controller.coffee [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    @pendingRequests.length == 0
+
```
Comment (HSalmon) [inline lineman/app/components/membership_requests_page/membership_requests_page_controller.coffee]: @noPendingRequests function doesn't seem to work - had to take it out of the view...

**Pull request merged**